File: MusicWeb.Server/MusicWeb.Models/Entities/Songs/Song.cs
using MusicWeb.Models.Entities.Artists;
using MusicWeb.Models.Entities.Base;
using MusicWeb.Models.Entities.Ratings;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicWeb.Models.Entities
{
    public class Song:BaseEntity
    {
        public string Name { get; set; }
        public DateTime ReleaseDate { get; set; }
        public double Length { get; set; }
        public string ImagePath { get; set; }
        public int PositionOnAlbum { get; set; }
        public string Text { get; set; }

        public int AlbumId { get; set; }
        public int ComposerId { get; set; }

        public virtual Album Album { get; set; }
        public virtual Artist Composer { get; set; }
        public virtual ICollection<SongGuestArtist> SongGuestArtists { get; set; }
        public virtual ICollection<UserFavoriteSong> UserFavoriteSongs { get; set; }
        public virtual ICollection<SongReview> SongReviews{ get; set; }
        public virtual ICollection<SongRating> SongRatings { get; set; }
    }
}

File: MusicWeb.Client/src/models/Song.js
export default class Song {
  id = null;
  name = "";
  releaseDate = "";
  duration = "";
  positionOnAlbum = "";
  description = "";
  albumId = "";
  composerId = "";
  album = {};
  composer = {};
  songReviews = [];
  rating = null;
  ratingsCount = null;
  favoriteCount = null;
  reviewsCount = null;
  albumName = "";
  artistName = "";

  constructor($data) {
    if ($data) {
      this.id = $data.id;
      this.name = $data.name;
      this.releaseDate = $data.releaseDate;
      this.duration = $data.length;
      this.positionOnAlbum = $data.positionOnAlbum;
      this.description = $data.text;
      this.albumId = $data.albumId;
      this.composerId = $data.composerId;
      this.albumName = $data.albumName;
      this.artistName = $data.artistName;
      this.album = $data.album ? $data.album : {};
      this.composer = $data.composer ? $data.composer : {};
      this.songReviews = $data.songReviews ? $data.songReviews : [];
      this.rating = $data.rating ? $data.rating : 0;
      this.ratingsCount = $data.ratingsCount ? $data.ratingsCount : 0;
      this.favoriteCount = $data.favoriteCount ? $data.favoriteCount : 0;
      this.reviewsCount = $data.reviewsCount ? $data.reviewsCount : 0;
    }
  }
}
